plotting/scriptGenerator.py

```diff
@@ -33,110 +25,40 @@ def time_query_conversion(dataset, index):
-    script.write("from urllib.parse import urlencode\n")
-    script.write("from contextlib import closing\n")
-    script.write("try:\n")
-    script.write("   from PIL import Image\n")
-    script.write("except:\n")   
-    script.write('   print("If you are on a Windows machine, please install PIL (imaging library) using' + " 'python -m pip install Pillow'" + ' in the Anaconda Prompt")\n')
-    script.write("   exit()\n")
-    script.write("import json\n")
-    script.write("\n\n")
-    #Set Navigators URL
-    script.write("#Set Navigator URL\n")
-    if not isClass4:
-        script.write('base_url = "http://navigator.oceansdata.ca/api/v1.0/%s/?"\n\n' % (data_type))
-    else:
-        script.write('base_url = "http://navigator.oceansdata.ca/%s/?"\n\n' % (data_type))
-    #---------------------------
-    
-        
-    #CREATE JSON OBJECT---------
-    script.write("#Create JSON Object\n")
-    script.write("query = {\n")
-    for x in url:
-      #print(type(url.get(x)))
-      if isinstance(url.get(x), str):
-        script.write('  "' + x + '": "' + str(url.get(x)) + '"' + ",\n" )
-      else:
-        
-        if x == 'time':
-            formatted_date = time_query_conversion(url.get('dataset'),url.get('time'))
-            script.write('  "' + x + '": ' + "'" + formatted_date + "',\n")
-        else:
-            script.write('  "' + x + '": ' + str(url.get(x)) + ",\n")
-    script.write("}\n")
-    #---------------------------
-    #Assemble full request
-    script.write('\n#Assemble full request - converts json object to url\n')
-    if notPlot == False:
-        script.write("url = base_url + urlencode(" + '{"query": ' + "json.dumps(query)})" + "\n")
+        var = "class4id"
+    elif "drifter" in url:
+        var = "drifter"
     else:
-        script.write("url = base_url + urlencode(" + '{"query": ' + "json.dumps(query)}) + '" + url_tail + "'\n")
+        var = "variable"
+    query = url
+
+    # format query for python, could do with regular expressions
+    query = query.replace(',"', ',\n"')
+    if "true" in query:
+        query = query.replace("true", "1")
+    if "false" in query:
+        query = query.replace("false", "0")
+    if "null"  in query:
+        query = query.replace("null", "None")
     
-    script.write("print(url)\n")
-    #Open URL and read response
-    if notPlot == False and netcdf == False:
-        script.write("\n#Open URL and save response\n")
-        script.write("with closing(urlopen(url)) as f:\n")
-        script.write("  img = Image.open(f)\n")
-        if url.get("type") == "drifter":
-            script.write('  img.save("script_template_" + str(query["dataset"]) + "_" + str(query["drifter"]) + ".png", "PNG")\n')
-        elif isClass4:
-            script.write('  img.save("script_template_" + str(query["dataset"]) + "_" + str(query["class4id"]) + ".png" , "PNG")\n')
+    with open("plotting/templates/python" + scriptType + "template.txt", 'r') as file:
```

Comment:
From codefactor: `Redefining built-in 'file' (redefined-builtin)`. perhaps using `f`?